private-sessions/README.md

```diff
@@ -0,0 +1,124 @@
+- Intel PSW (Platform SoftWare): (Note: this is already done on Azure SGX VMs). This provides several SGX functionalities from loading and initializing SGX enclaves to management of so-called "architectural" enclaves. It runs as a Linux service (aesm_service for Application Enclave Services Manager). For information, the interface to this service is through a socket located at /var/run/aesmd/aesm.socket (this will be usefull later, because we need to provide this socket to the server container). TODO: check if we had to install this + link to instructions
+- Intel DCAP library: This binary library provides functionality for dcap quote generation and verification. TODO check if we had to install this + instructions
+- Intel QPL library: This binary library provides functionality for the communication with DCAP verification services. TODO add qpl config + difference with azure_dcap library.
+
+At this stage, it is highly recommended to check the setup by running the python `sgx-quote.py` example from the Gramine repository (see [here](https://github.com/gramineproject/gramine/tree/master/CI-Examples/python)). This will also help understand more about how an application is "graminized", The Gramine `is-sgx-available` can also be of good use.
+
+### Gramine Shielded Containers
+
+This tool is used to "graminize" a standard docker image so that its content can be executed within an SGX enclave.
+
+If the reader has already played with the Gramine CI-Examples linked above, it should have some understanding of how to configure the process of "graminizing" an application with the manifest file and how to run it with Gramine-SGX. GSC automates this entire process for existing Docker images by creating a manifest file with the correct configuration (entrypoint, sgx trusted files, etc.) and a new image containing the required application files, a Gramine install as well as the signed enclave.
+
+The GSC tool comes as a python script. It is installed by cloning [this](https://github.com/gramineproject/gsc) Git repository. Documentation for the tool as well as installation instructions can be found [here](https://gramine.readthedocs.io/projects/gsc/en/latest/).
+
+
+## Python package for gramine ra-tls related functionality
+
+Note: the package is currently located in ./src/gramine_ratls but we could consider publishing it as an independent package for
+others to use.
+
+As stated above, the server uses ra-tls to attest its code and runing environment. Before starting the fastapi app, the python server must create the ra-tls certificate and corresponding key using Gramine ra-tls binary libraries and write them to disk. These will be used by uvicorn as server certificate and provide the attestation to the client.
```

Comment:
I am a bit confused, is this the same key as before or a new one ?

Reply:
I think it is and it creates the key but I don't understand what is the point of the key